File: calendarApp/ClientApp/src/types/ApplicationUser.ts
import { Calendar } from './Calendar'

export interface ApplicationUser {
  UserId: string
  Email: string
  FirstName: string
  LastName: string
  Password: string
  Role: string

  Calendars: Calendar[]
}

export interface UserLoginInput {
  Email: string
  Password: string
}

export interface UserRegisterInput {
  Email: string
  FirstName: string
  LastName: string
  Password: string
}

File: calendarAppAPI/Models/DbModels/ApplicationUser.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Globalization;
using Microsoft.AspNetCore.Identity;

namespace CalendarApp.Models.DbModels;

[Table ("Users")]
public class ApplicationUser : IdentityUser {
	[Key]
	public string UserId { get; set; }
	public string? Email { get; set; }
	public string? FirstName { get; set; }
	public string? LastName { get; set; }
	public string? Password { get; set; }
	public string? Role { get; set; }

	// Relational Data
	public virtual IList<Calendar> Calendars { get; set; } = new List<Calendar> ();
}